common/components/forms/ContactForm.jsx

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+import React from 'react';
+import ProjectAPIUtils from '../utils/ProjectAPIUtils.js';
+import ReCAPTCHA from "react-google-recaptcha";
+import _ from 'lodash';
+
+class ContactForm extends React.Component {
```

Comment:
I agree with the last three but I've got reservations about the first. Nested state has some real pitfalls for performance and code cleanliness; see [this Stackoverflow thread](https://stackoverflow.com/a/51136076) for example. Generally I'm 100% on board with you that we need to be consistent in pattern but this one I'd hesitate on.

Reply:
Okay, you can keep the form fields directly in state.  A couple tips:
- If you want to separate out the form fields in flow to keep things straight, you can concatenate two types together and do something like type State = FormFields & OtherStateVariablesThatAreNotFormFields
- If you don't use a formFields state variable, make sure to pass in a custom getter and setter into form.setup().

Reply:
I took some time to step back and think about this while researching some of the errors I'm getting, and I am no longer convinced we should be refactoring this page to use the Create Project form model. 

First, code reuse: I do not feel I am reusing enough of the Create Project form code. Custom form getter and setter, different view, different needs on the backend (submit a request to Google and perform an email action based on that, not save to the db) and I'm not using any of the form_helper functions. I have little doubt when I get around to validation that's going to be custom as well. To me this defeats the purpose of trying to reuse the Create Project form code, because I'm _not_ reusing most of it, I'm writing a parallel version of it.

Second, conceptually, I think this page is far closer to the pattern for contacting a project or a project leader being able to contact their volunteers. Take this Contact Project component:
<img width="513" alt="Screen Shot 2019-08-22 at 11 25 17 AM" src="https://user-images.githubusercontent.com/16870466/63539961-09502600-c4d0-11e9-8128-23803ebc6382.png">

Take that out of a modal and you have the Contact Us form - fill out some formfields and an email is sent on your behalf. I think this should be the pattern I follow.

Admittedly, some of my thinking is that I originally patterned this PR on the contact volunteer/contact project model and it worked and was pretty lightweight in code terms and is no longer either of those things, but I don't think that invalidates my position.

So, long story short, I propose reverting to the version of this PR which was closely patterned on the contact volunteer/contact project email pattern rather than the create project pattern because the purpose is to generate an email and send it on behalf of the user, just like those components. What do you think?